src/NewTools-FileBrowser/StDirectoryTreePresenter.class.st

```diff
@@ -49,11 +49,13 @@ StDirectoryTreePresenter >> expandPath: aFileLocator [
 	"Expand the receiver's tree to aFileLocator reference"
 
 	| path aPathForSpec currentNode |
-	path := aFileLocator asPath segments asOrderedCollection.
-	aPathForSpec := OrderedCollection new.
-	aPathForSpec add: 1.
+	path := aFileLocator absolutePath segments asOrderedCollection.
+	aPathForSpec := OrderedCollection with: 1.
 
 	currentNode := directoryTreePresenter roots anyOne.
+	Smalltalk os isWindows ifTrue: [
```

Comment:
Jan what do think about the following:
- could we pass the os (self class environment os) to the presenter when it is instantiated. Like that we do not have this global variable right in the middle. 
- Potentially later we could have a strategy for each os but this is another step.

Reply:
> could we pass the os (self class environment os) to the presenter when it is instantiated. Like that we do not have this global variable right in the middle.

You are right, I am, however, failing to find a way to do that without hampering usability. Problem is that most of these presenters need to know the OS before `initializePlatform`, therefore before `initialize`, and there are many methods sending it.
`new, owner:, newApplication:, newApplication:model:, newApplication:owner: newApplication:owner:model:, owner:on:`, these all are senders of `initialize`, so they could be used and would need to have a variant with another parameter (the platform) instead.
Maybe to not actually show the contents until the platform is set and then reload the layout? IDK

Reply:
Shouldn't the SystemEnvironment have an `os` method?
I would let this to be addressed in another issue, since it is part of a wider problemn that could need other integrations before.